packages/content/src/Application/ResourceLoader/Loader/CachedResourceLoader.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Content\Application\ResourceLoader\Loader;
+
+use Symfony\Contracts\Service\ResetInterface;
+
+class CachedResourceLoader implements ResourceLoaderInterface, ResetInterface
+{
+    /**
+     * @var array<string, array<int|string, mixed>>
+     */
+    private array $cache = [];
+
+    public function __construct(private ResourceLoaderInterface $decoratedResourceLoader)
+    {
+    }
+
+    public function load(array $ids, ?string $locale, array $params = []): array
+    {
+        $cacheKey = $this->generateCacheKey($ids, $locale, $params);
+
+        if (isset($this->cache[$cacheKey])) {
+            return $this->cache[$cacheKey];
+        }
+
+        $result = $this->decoratedResourceLoader->load($ids, $locale, $params);
+        $this->cache[$cacheKey] = $result;
+
+        return $result;
+    }
```

Comment:
This looks unexpected we do NOT use here a memoize based cache mechanic here.

We need cache here by `id` (single) and merge it for next call. We can have first call with `[1, 2]` second with `[1, 3]` then the decorated is only called with `[3]`.:

```suggestion
    public function load(array $ids, ?string $locale, array $params = []): array
    {
        $notFoundIds = [];
        $cachedResult = [];
        foreach ($ids as $id) {
            $cacheKey = $id;
            if (!isset($this->cache[$cacheKey])) {
                 $notFoundIds = [];
                 
                 continue;
            }
            
            $cachedResult[$id] = $this->cache[$cacheKey];
        }
        
        $result = $this->decoratedResourceLoader->load($ids, $locale, $params);
        
        // cache new result
        // merge with cached result
        
        return $result;
    }
```